R/sort_linter.R

```diff
@@ -0,0 +1,86 @@
+#' Require usage of `sort()` over `.[order(.)]`
+#'
+#' `sort()` is the dedicated option to sort a list or vector. It is more legible
+#' and around twice as fast as `.[order(.)]`, with the gap in performance
+#' growing with the vector size.
```

Comment:
Something worth mentioning here, a use case like:

```r
y <- y[order(x)]
x <- x[order(x)]
```

Where we are sorting multiple objects according to the order in one "master" reference. In that case, the recommendation is something like:

```r
index <- order(x)
y <- y[index]
x <- x[index]

# above is better than
y <- y[order(x)]
x <- sort(x)
```

WDYT? Should we make some reference to this use case in the lint message too?

Reply:
You're right, I've added an example but I believe the lint message is already complicated enough so we should not add more information.